src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -157,6 +162,11 @@ - (instancetype)initWithHost:(NSString *)host
     _requestWriter = requestWriter;
 
     _requestHeaders = [[GRPCRequestHeaders alloc] initWithCall:self];
+
+    if ([requestWriter isKindOfClass:[GRXImmediateSingleWriter class]]) {
+      _unaryCall = true;
+      _unaryOpBatch = [[NSMutableArray alloc] init];
```

Comment:
nit: `[NSMutableArray array]` seems to be the preferred style. In this case, you even know the number of elements it'll have, right? So you can do `[NSMutableArray arrayWithCapacity:...]`.